branchpro/models.py

```diff
@@ -42,3 +43,106 @@ def simulate(self, parameters, times):
             value appears in ``times``.
         """
         raise NotImplementedError
+
+#
+# BranchProModel Class
+#
+
+
+class BranchProModel(ForwardModel):
+    """BranchProModel Class:
+    Class for the models following a Branching Processes behaviour.
+    It inherits from the `ForwardModel`` class.
+
+    Parameters
+    ----------
+    initial_r
+        (numeric) Value of the recombination number at the beginning
+        of the epidemic
+    serial_interval:
+        (list) Unnormalised probability distribution of that the recipient
+        first displays symptoms s days after the infector first displays
+        symptoms
+
+    Methods
+    -------
+    simulate: return model output for specified parameters and times.
+    _normalised_daily_mean: (Private) returns the expected number of new cases
+        at time t
+    """
+
+    def __init__(self, initial_r, serial_interval):
+        super(BranchProModel, self).__init__()
+
+        if not isinstance(serial_interval, list):
+            raise TypeError('Serial interval values must be in a list format')
+        if np.sum(serial_interval) <= 0:
+            raise ValueError('Sum of serial interval values must be > 0')
+        if not isinstance(initial_r, (int, float)):
+            raise TypeError('Value of R must be integer or float')
+
+        self._serial_interval = np.asarray(serial_interval)[::-1]
+        self._initial_r = initial_r
+        self._normalizing_const = np.sum(self._serial_interval)
+
+    def _normalised_daily_mean(
+            self, t, incidences):
+        """
+        Computes xpected number of new cases at time t, using previous
+        incidences data and serial intevals.
```

Comment:
"incidences and serial intervals" is better. Data would probably make people think that you have incidence observations.